src/wp-includes/meta.php

```diff
@@ -1459,6 +1463,13 @@ function register_meta( $object_type, $meta_key, $args, $deprecated = null ) {
 		}
 	}
 
+	// If the object_type is set, require a type that supports revisions when setting revisions_enabled to true.
+	if ( ! empty( $object_type ) && $args['revisions_enabled'] && ! post_type_supports( $object_type, 'revisions' ) ) {
+		_doing_it_wrong( __FUNCTION__, __( 'Meta keys cannot enable revisions support unless the object type supports revisions.' ), '6.4.0' );
+
+		return false;
+	}
+
 	$object_subtype = ! empty( $args['object_subtype'] ) ? $args['object_subtype'] : '';
```

Comment:
```suggestion
	$object_subtype = ! empty( $args['object_subtype'] ) ? $args['object_subtype'] : '';
	// If the object_type is set, require a type that supports revisions when setting revisions_enabled to true.
	if ( ! empty( $object_subtype ) && $args['revisions_enabled'] && ! post_type_supports( $object_subtype, 'revisions' ) ) {
		_doing_it_wrong( __FUNCTION__, __( 'Meta keys cannot enable revisions support unless the object type supports revisions.' ), '6.4.0' );

		return false;
	}
```

Reply:
I'm not sure this works. If the object type is empty (""), the meta is for ALL post types, but your logic will throw doing_it_wrong in this case.

Reply:
@adamsilverstein To be clear, there is object type and $object_subtype. Object type can be user, post, term etc. Sub type is for post types, like page. 

The two checks are designed to do this. 

- If user try assign `revisions_enabled` enabled on say users, return a doing it wrong. 
- If type of post is set and object subtype is set and subtype doesnt support revisions, return a doing it wrong.

Reply:
Oh, thanks for clarifying! I totally missed that you were checking the subtype when reviewing the suggestion here on GitHub.